### DISCUSSION TITLE
What type I have to choose if I don't need timezone? What about current time?

### DISCUSSION THREAD

**Discussion by frederikhors** [SeaQL/sea-orm#1053] [Q&A] [answered]
Coming from Go I was using this native type for the created_at/updated_at fields:

```go
type Spv struct {
	createdAt        time.Time
	balanceUpdatedAt *time.Time
}
```

Now with Rust I'm using the `.timestamp()` method in migrations which creates in Postgres tables with columns like:

```sql
created_at timestamp without time zone DEFAULT CURRENT_TIMESTAMP NOT NULL,
updated_at timestamp without time zone,
```

and in generated rust code fields like:

```rust
pub created_at: TimeDateTime,
pub updated_at: Option<TimeDateTime>,
```

which I love! :heart:

Now in my code I choose to use `time` over `chrono` because I need to achieve simple operations.

What I don't understand is what type to use in my domain structs:

```rust
use time::{PrimitiveDateTime};

pub struct Spv {
  pub created_at: PrimitiveDateTime,
  pub balance_updated_at: Option<PrimitiveDateTime>,
}
```

If I use `PrimitiveDateTime` [I don't know how to create a variable with current time: it doesn't have a method called `now()`](https://stackoverflow.com/questions/73790383/how-to-assign-to-a-new-variable-the-current-time-using-timeprimitivedatetime-t).

If I use `OffsetDateTime` it has `now_utc()` which I can use but I don't know if this is correct and if I can easily convert one to another.

Given I don't need timezone, what do you suggest to use?

Thanks in advance.

**Comment by billy1624** [comment]
Hey @frederikhors, I'd suggest using `PrimitiveDateTime` given that you don't need timezone support.

On how to create a `PrimitiveDateTime` of current time. I think someone answer you on StackOverflow?

**Comment by frederikhors** [comment, marked as answer]
I decided to use timestamps with timezone,
